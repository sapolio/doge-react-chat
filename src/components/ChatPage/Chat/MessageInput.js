import React from 'react';
import { withStyles } from 'material-ui/styles';
import Paper from 'material-ui/Paper';
import Input from 'material-ui/Input';
import Button from 'material-ui/Button';

const styles = theme => ({
  messageInputWrapper: {
    position: 'fixed',
    left: 320,
    right: 0,
    bottom: 0,
    padding: theme.spacing.unit * 3,
  },
  messageInput: {
    padding: theme.spacing.unit * 2,
  },
  button: {
    width: '100%',
  },
});

class MessageInput extends React.Component {
  render() {
    const { classes, showJoinButton, onJoinButtonClick } = this.props;

    return (
      <div className={classes.messageInputWrapper}>
        <Paper className={classes.messageInput} elevation={10}>
          {showJoinButton ? (
            <Button
              variant="raised"
              color="primary"
              className={classes.button}
              onClick={onJoinButtonClick}>
            So Much Join Very Chat
            </Button>
          ) : (
              <Input fullWidth placeholder="Type your messageâ€¦" />)}
        </Paper>
      </div>
    );
  }
}

export default withStyles(styles)(MessageInput);
