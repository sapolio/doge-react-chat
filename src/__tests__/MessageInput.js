/* eslint-env jest */
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import MessageInput from '../components/ChatPage/Chat/MessageInput';

const props = {
  showJoinButton: false,
  disabled: false,
  onJoinButtonClick: jest.fn(),
  sendMessage: jest.fn(),
};

describe('<MessageInput />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<MessageInput {...props} />, div);
    ReactDOM.unmountComponentAtNode(div);
    ReactDOM.render(<MessageInput {...props} showJoinButton disabled />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  it('renders no button', () => {
    const tree = renderer.create(<MessageInput {...props} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders no button disabled', () => {
    const tree = renderer.create(<MessageInput {...props} disabled />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders with button', () => {
    const tree = renderer.create(<MessageInput {...props} showJoinButton />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders with button disabled', () => {
    const tree = renderer.create(<MessageInput {...props} showJoinButton disabled />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
