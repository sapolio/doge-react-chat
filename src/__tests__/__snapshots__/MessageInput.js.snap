// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MessageInput /> renders no button 1`] = `
<div
  className="MessageInput-messageInputWrapper-93"
>
  <div
    className="MuiPaper-root-96 MuiPaper-shadow10-108 MuiPaper-rounded-97 MessageInput-messageInput-94"
  >
    <div
      className="MuiInput-root-123 MuiInput-fullWidth-131 MuiInput-inkbar-125 MuiInput-underline-129"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyPress={[Function]}
    >
      <input
        aria-invalid={undefined}
        aria-required={undefined}
        autoComplete={undefined}
        autoFocus={undefined}
        className="MuiInput-input-132"
        defaultValue={undefined}
        disabled={false}
        id={undefined}
        name={undefined}
        onChange={[Function]}
        onKeyDown={undefined}
        onKeyUp={undefined}
        placeholder="Type your WOW"
        readOnly={undefined}
        required={undefined}
        rows={undefined}
        type="text"
        value=""
      />
    </div>
  </div>
</div>
`;

exports[`<MessageInput /> renders no button disabled 1`] = `
<div
  className="MessageInput-messageInputWrapper-93"
>
  <div
    className="MuiPaper-root-96 MuiPaper-shadow10-108 MuiPaper-rounded-97 MessageInput-messageInput-94"
  >
    <div
      className="MuiInput-root-123 MuiInput-disabled-128 MuiInput-fullWidth-131 MuiInput-inkbar-125 MuiInput-underline-129"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyPress={[Function]}
    >
      <input
        aria-invalid={undefined}
        aria-required={undefined}
        autoComplete={undefined}
        autoFocus={undefined}
        className="MuiInput-input-132 MuiInput-inputDisabled-134"
        defaultValue={undefined}
        disabled={true}
        id={undefined}
        name={undefined}
        onChange={[Function]}
        onKeyDown={undefined}
        onKeyUp={undefined}
        placeholder="Type your WOW"
        readOnly={undefined}
        required={undefined}
        rows={undefined}
        type="text"
        value=""
      />
    </div>
  </div>
</div>
`;

exports[`<MessageInput /> renders with button 1`] = `
<div
  className="MessageInput-messageInputWrapper-93"
>
  <div
    className="MuiPaper-root-96 MuiPaper-shadow10-108 MuiPaper-rounded-97 MessageInput-messageInput-94"
  >
    <button
      className="MuiButtonBase-root-153 MuiButton-root-138 MuiButton-raised-143 MuiButton-raisedPrimary-145 MessageInput-button-95"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label-139"
      >
        So Much Join Very Chat
      </span>
      <span
        className="MuiTouchRipple-root-155"
      />
    </button>
  </div>
</div>
`;

exports[`<MessageInput /> renders with button disabled 1`] = `
<div
  className="MessageInput-messageInputWrapper-93"
>
  <div
    className="MuiPaper-root-96 MuiPaper-shadow10-108 MuiPaper-rounded-97 MessageInput-messageInput-94"
  >
    <button
      className="MuiButtonBase-root-153 MuiButtonBase-disabled-154 MuiButton-root-138 MuiButton-raised-143 MuiButton-raisedPrimary-145 MuiButton-disabled-147 MessageInput-button-95"
      disabled={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={-1}
      type="button"
    >
      <span
        className="MuiButton-label-139"
      >
        So Much Join Very Chat
      </span>
    </button>
  </div>
</div>
`;
